include/deal.II/base/geometry_info.h

```diff
@@ -1589,7 +1921,7 @@ struct GeometryInfo
    * with the UCD numbering, this field can also be used like a
    * old_to_lexicographic mapping.
    */
-  static const unsigned int ucd_to_deal[vertices_per_cell];
+  static constexpr std::array<unsigned int, vertices_per_cell> ucd_to_deal = internal::initialize_ucd_to_deal<dim>::value();
```

Comment:
and then initializing here not by the function call, but by just the array that is a public member of `internal::initialize_ucd_to_deal<dim>`?

This would require the compiler fewer headstands to see through the control flow and make constant propagation easier.